### PR TITLE
[generator] Bug Fix: Restrict markdown table parser to child rows and expand edge-case test coverage

### PR DESCRIPTION
**Pull request by jwren**

### CODE REVIEW

File: tool/generator/lib/src/services/markdown_converter.dart [hunk 1]
```diff
@@ -141,46 +141,50 @@ class MarkdownConverter {
...
+      headerCells = firstRow.children
+          .where((e) => e.localName == 'th')
+          .toList();
       if (headerCells.isEmpty) {
-        headerCells = firstRow.querySelectorAll('td');
+        headerCells = firstRow.children
+            .where((e) => e.localName == 'td')
+            .toList();
       }
       if (bodyRows.isNotEmpty) bodyRows.removeAt(0);
```
[the remark this answers is not in the document]
Comment (jwren) [reply tool/generator/lib/src/services/markdown_converter.dart]: resolved.

File: tool/generator/lib/src/services/markdown_converter.dart [hunk 2]
```diff
@@ -141,46 +141,50 @@ class MarkdownConverter {
...
+        .where((e) => e.localName == 'tbody')
+        .toList();
+    if (tbodies.isNotEmpty) {
+      for (final tbody in tbodies) {
+        bodyRows.addAll(tbody.children.where((e) => e.localName == 'tr'));
       }
+    } else {
+      // No tbody, check direct tr children
+      bodyRows.addAll(table.children.where((e) => e.localName == 'tr'));
     }
```
[the remark this answers is not in the document]
Comment (jwren) [reply tool/generator/lib/src/services/markdown_converter.dart]: resolved.
